File: server/src/scraper/entities/character_entity.ts
import {Column, Entity, JoinColumn, ManyToMany, ManyToOne, OneToMany, PrimaryGeneratedColumn} from "typeorm";
import {CharElementEntity} from "./char_element.entity";
import {CharacterLevelEntity} from "./character_level_entity";

@Entity()
export class CharacterEntity {
    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    name: string

    @Column()
    href: string

    @Column({nullable: true})
    rarity: number
    
    @Column()
    imageUrl: string

    @Column({nullable: true})
    mainImageUrl: string

    @Column({nullable: true})
    buble: string

    @Column({nullable: true})
    description: string

    @Column({nullable: true})
    birthday: string

    @Column({nullable: true})
    introduction: string

    //stats

    @ManyToOne(() => CharElementEntity, character => character.characters)
    characterElement: CharElementEntity;

    @OneToMany(() => CharacterLevelEntity, level => level.character)
    levels: CharacterLevelEntity[];


}

File: mobile/lib/domains/character/character_entity.dart


import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:mobile/domains/character/character_level_entity.dart';
import 'package:mobile/domains/character_element/character_element_entity.dart';

part 'character_entity.freezed.dart';
part 'character_entity.g.dart';
@freezed
class CharacterEntity with _$CharacterEntity {
  const factory CharacterEntity({
    int? id,
    String? name,
    String? href,
    int? rarity,
    String? imageUrl,
    String? mainImageUrl,
    String? buble,
    String? description,
    String? birthday,
    String? introduction,
    CharacterElementEntity? characterElement,
    @JsonKey(name: "levels")
    List<CharacterLevelEntity>? characterLevels,
  }) = _CharacterEntity;

  factory CharacterEntity.fromJson(Map<String, dynamic> json) =>
      _$CharacterEntityFromJson(json);

}